src/main/java/cmu/pasta/mu2/diff/guidance/OptimizedMutationGuidance.java

```diff
@@ -0,0 +1,7 @@
+package cmu.pasta.mu2.diff.guidance;
+
+import cmu.pasta.mu2.util.ArraySet;
+
+public interface OptimizedMutationGuidance extends DiffGuidance {
```

Comment:
This interface need not extend `DiffGuidance`, since it has nothing to do with differential fuzzing. The mutant filtering is independent and only an optimization for which mutants to run; this can be combined with differential fuzzing, which is a test oracle.

Reply:
On second thoughts, I don't think we need this interface to be a `Guidance` at all. Would it better to have a `MutantFilter` interface instead of an `OptimizedMutationGuidance`? Perhaps we can have filters that implement this interface and then just selectively choose which filters to apply from a single `MutationInstance`. Right now, we have way too many subclasses of `MutationGuidance`, and we cannot compose the various filters.